codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -125,23 +128,23 @@ private static ByteBuf encodeConnectMessage(
         if (!isValidClientId(mqttVersion, clientIdentifier)) {
             throw new MqttIdentifierRejectedException("invalid clientIdentifier: " + clientIdentifier);
         }
-        byte[] clientIdentifierBytes = encodeStringUtf8(clientIdentifier);
-        payloadBufferSize += 2 + clientIdentifierBytes.length;
+        int clientIdentifierBytes = utf8Bytes(clientIdentifier);
+        payloadBufferSize += 2 + clientIdentifierBytes;
 
         // Will topic and message
         String willTopic = payload.willTopic();
-        byte[] willTopicBytes = willTopic != null ? encodeStringUtf8(willTopic) : EmptyArrays.EMPTY_BYTES;
+        int willTopicBytes = nullableUtf8Bytes(willTopic);
         byte[] willMessage = payload.willMessageInBytes();
         byte[] willMessageBytes = willMessage != null ? willMessage : EmptyArrays.EMPTY_BYTES;
```

Comment:
Above lines could be moved inside the `if` block I think? (I know not strictly related to this PR)